index.bs

```diff
@@ -1078,38 +1095,150 @@ Issue: Define how this works.
 Each EMCAScript [=Realm=] has a corresponding <dfn>handle object map</dfn>. This is a
 strong map from handle ids to their corresponding objects.
 
+Each {{Window}} has a corresponding <dfn>platform objects map</dfn>. This is a weak
+map from shared ids to their corresponding [=platform object=].
+
 <pre class="cddl remote-cddl">
 OwnershipModel = "root" / "none";
 </pre>
 
-<dfn>RemoteReference</dfn> represents a remote reference to an existing ECMAScript
-object in [=handle object map=] in the given [=Realm=].
+<dfn>RemoteReference</dfn> is either a <code>RemoteObjectReference</code>
+representing a remote reference to an existing ECMAScript object in
+[=handle object map=] in the given [=Realm=], or is a <code>SharedReference</code>
+representing a reference to a [=platform object=] in the given [=browsing context=]'s
+[=list of known elements=], [=list of known shadow roots=] on in the given
+{{Window}}'s [=platform objects map=].
 
 <pre class="cddl remote-cddl local-cddl">
+RemoteReference = {
```

Comment:
Could we just have a single variant that always has a `handle` and optionally has a `sharedId`? That has a few advantages:
* Some client functionality can just unconditionally use `handle` and not care about `sharedId`.
* It makes it easier to add a sharedId to new DOM types in the future if we need to, without breaking working code.
* We don't have to introspect the remote reference object to figure out which variant we got. If we do keep this design we need to add a `type` field.

The main disadvantage I can think of is that we might really want to encourage people to always use `sharedId` if we have one.

Reply:
I don't quite understand the suggestion. I understood it as "let's make the handle field mandatory", which basically means the `sharedId` is useless. Maybe there is a confusion between "RemoteReference" and "RemoteValue"? The first is passed to server by client, the last is sent by server to client.